### DISCUSSION TITLE
Lazer has broken skin import?

### DISCUSSION THREAD

**Discussion by DarkGradient** [ppy/osu#28565] [Q&A] [answered]
For some reason, I'm having strange problems importing my skin from stable release into lazer. The 6k modes for mania and above look very strange, like my assets are lost, but they are not, I tried to change the skin structure itself, but that didn't help either.

I have attached screenshots below with more information.
![osu!_QSlsqK0hhP](https://github.com/ppy/osu/assets/82389028/c1ae07c4-42dc-4d3c-921d-7c37041e4963)
![osu!_F2JCEPXDEW](https://github.com/ppy/osu/assets/82389028/198bde93-e979-4dee-9ea2-fcbf1bfbe483)

**Comment by peppy** [comment]
Could you please attach your skin?

**Reply by DarkGradient** [reply]
sure
(file too big, only can send via gdrive)
[Skin link](https://drive.google.com/file/d/12dVaBFxuhpQDOXKscEqUB061OB93zM6z/view?usp=sharing)

**Reply by peppy** [reply]
Okay, so your skin configuration only has configurations for 1K and 2K.

If I test on either of these key counts in lazer it works fine.

![osu! 2024-06-27 at 09 34 44](https://github.com/ppy/osu/assets/191335/3a206f7f-e442-4d93-8d12-8c5bf79cc620)

If you want an immediate fix, you should copy the configuration across to other key counts too.

**Reply by DarkGradient** [reply]
huh?
my skin only has 1k and 2k support?, that isn't true.
![image](https://github.com/ppy/osu/assets/82389028/f16bdf1a-ae9a-4d6f-955d-1d4e9486d3f4)
My skin supports up to 9k, but it doesn't work.

**Reply by bdach** [reply, marked as answer]
The above isn't correct. What is *actually* going on here is that your skin has half-broken `ColourLight` specs for 6K and above:

```
[Mania]
Keys: 6
// ...
ColourLight: 255,255,255,255,255,255,255,255

[Mania]
Keys: 7
// ...
ColourLight: 255,255,255,255,255,255,255,255,255

[Mania]
Keys: 8
// ...
ColourLight: 255,255,255,255,255,255,255,255,255,255

[Mania]
Keys: 9
// ...
ColourLight: 255,255,255,255,255,255,255,255,255,255,255
```

lazer expects at most four numbers here (as it should - R/G/B/A) and stops parsing the rest of the section if it finds broken stuff. stable probably just parses the first four and ignores the rest.

One way to fix would be to fix the skin to not have broken colours. Not sure we want to be adjusting our code for this?

**Reply by peppy** [reply]
Weird, somehow the `skin.ini` got cut off when I opened it.

> One way to fix would be to fix the skin to not have broken colours. Not sure we want to be adjusting our code for this?

I'm not sure it's worth fixing, yeah.

**Reply by DarkGradient** [reply]
oh, you are right, i have some broken lines.
thanks for the tip on the broken line
(idk how it still working fine in stable)
